packages/next/build/webpack-config.ts

```diff
@@ -1001,14 +1002,49 @@ export default async function getBaseWebpackConfig(
       return `commonjs next/dist/lib/import-next-warning`
     }
 
+    const resolveWithReactServerCondition =
+      layer === WEBPACK_LAYERS.server
+        ? getResolve({
+            alias: process.env.__NEXT_REACT_CHANNEL
```

Comment:
This gonna expose `__NEXT_REACT_CHANNEL` to users, we should rely on the require-hook to map the react and react dom package exports. `__NEXT_REACT_CHANNEL` is only used for testing now

Reply:
@huozhi Why would this expose `__NEXT_REACT_CHANNEL` to users?

Reply:
I mean now you can pass `__NEXT_REACT_CHANNEL` as env vars and can affect webpack configs, but previously it won't.

Reply:
I'm not now I understand "you can pass" and "gonna expose" here. You can do anything to break Next.js but I'm not gonna do that. ;)

Reply:
I was referring to leverage the existing alias, the change looks good.